module/VuFind/src/VuFind/View/Helper/Root/Icon.php

```diff
@@ -0,0 +1,151 @@
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\View\Helper\AbstractHelper;
+use Laminas\View\Helper\EscapeHtmlAttr;
+use VuFindTheme\ThemeInfo;
+
+/**
+ * Icon view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Icon extends AbstractHelper
+{
+    /**
+     * Icon config from theme.config.php
+     *
+     * @var array
+     */
+    protected $config;
+
+    /**
+     * Default icon set
+     *
+     * @var string
+     */
+    protected $defaultSet;
+
+    /**
+     * Transforming map
+     *
+     * @var array
+     */
+    protected $iconMap;
+
+    /**
+     * Constructor
+     *
+     * @param ThemeInfo $themeInfo Theme info helper
+     */
+    public function __construct(ThemeInfo $themeInfo)
+    {
+        $this->config = $themeInfo->getMergedConfig('icons', true);
+        $this->defaultSet = $this->config['defaultSet'] ?? 'FontAwesome';
+        $this->iconMap = $this->config['aliases'] ?? [];
+    }
+
+    /**
+     * Map icon to set. Add prefix, return with set and template
+     * Broken out for easier customization
+     *
+     * @param string $name Which icon?
+     *
+     * @return array
+     */
+    protected function mapIcon(string $name): array
+    {
+        $icon = $this->iconMap[$name] ?? $name;
+        $set = $this->defaultSet;
+
+        // Override set from config (ie. FontAwesome:icon)
+        if (strpos($icon, ':') !== false) {
+            [$set, $icon] = explode(':', $icon, 2);
+        }
+
+        // Find set in theme.config.php
+        $setConfig = $this->config['sets'][$set] ?? [];
+        $template = $setConfig['template'] ?? $set;
+        $prefix = $setConfig['prefix'] ?? '';
+
+        return [$prefix . $icon, $set, $template];
+    }
+
+    /**
+     * Reduce extra parameters to one attribute string
+     * Broken out for easier customization
+     *
+     * @param array          $extra   Just extra HTML attributes for now
+     * @param EscapeHtmlAttr $escAttr EscapeHtmlAttr view helper
+     *
+     * @return string
+     */
+    protected function compileAttrs(array $extra, EscapeHtmlAttr $escAttr): string
+    {
+        $attrs = '';
+        foreach ($extra as $key => $val) {
+            $attrs .= ' ' . $key . '="' . $escAttr($val) . '"';
+        }
+        return $attrs;
+    }
+
+    /**
+     * Returns inline HTML for icon
+     *
+     * @param string $name  Which icon?
+     * @param array  $extra Just extra HTML attributes for now
+     *
+     * @return string
+     */
+    public function __invoke(string $name, $extra = []): string
```

Comment:
This method does potentially quite a bit of work and could use some caching of results. Otherwise e.g. the format icons on a results list can cause quite a bit of repeated work.